cloud/services/domains.go

```diff
@@ -85,46 +82,47 @@ func DeleteIPFromDNS(ctx context.Context, mscope *scope.MachineScope) error {
 	domainHostname := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier
 
 	// Delete A record
-	for ipType, publicIP := range publicIPs {
-		if err := DeleteDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, ipTypeToRecordTypeMapper[ipType]); err != nil {
-			return err
+	for _, publicIPs := range publicIPMapper {
+		for ipType, publicIP := range publicIPs {
+			if err := DeleteDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, ipTypeToRecordTypeMapper[ipType]); err != nil {
+				return err
+			}
+			// Delete TXT record
+			if err := DeleteDomainRecord(ctx, mscope, domainHostname, publicIP, dnsTTLSec, domainID, "TXT"); err != nil {
+				return err
+			}
 		}
 	}
 
-	// Delete TXT record
-	txtRecordValueString := createSHA256HashOfString(mscope.LinodeMachine.Name)
-	if err := DeleteDomainRecord(ctx, mscope, domainHostname, "owner:"+txtRecordValueString, dnsTTLSec, domainID, "TXT"); err != nil {
-		return err
-	}
-
 	// Wait for TTL to expire
 	time.Sleep(time.Duration(dnsTTLSec) * time.Second)
 
 	return nil
 }
 
 // getMachinePublicIPs gets the machines public IP
-func getMachinePublicIPs(ctx context.Context, mscope *scope.MachineScope) (map[string]string, error) {
-	// Verify instance id is not nil
-	if mscope.LinodeMachine.Spec.InstanceID == nil {
-		return nil, errors.New("instance ID is nil. cant get machine's public ip")
-	}
-
-	// Get the public IP that was assigned
-	addresses, err := mscope.LinodeClient.GetInstanceIPAddresses(ctx, *mscope.LinodeMachine.Spec.InstanceID)
-	if err != nil {
-		return nil, err
-	}
-
-	if len(addresses.IPv4.Public) == 0 || addresses.IPv6 == nil {
-		return nil, errors.New("no public address")
-	}
-
-	if addresses.IPv6.SLAAC == nil {
-		return nil, errors.New("no SLAAC address")
+func getMachinePublicIPs(mscope *scope.MachineScope) ([]map[string]string, error) {
```

Comment:
Why does the return value need to be a slice of maps?

Reply:
DNS record type needs to be set based on IP type. So we nede to link IPv4/IPv6 to A/AAAA. Maps allows us to do that with the DNS record type.
If we use variables as return values, we need to do a isIPv4 or not check again when we try to determine what record needs to be created
The slice allows us to return something like [{"ipv4": "10.10.10.10"}, {"ipv6": "fd00::"}] and then reference returned_value["ipv4"] to set record type

Reply:
What I was thinking was something along the lines of this:

```go
func getMachinePublicIPs(mscope *scope.MachineScope) (ip4, ip6 netip.Addr, err error)
```

I don't think you would need to do another `ipIPv4` check again afterwards if you went this route. The individual values are named for the data they would hold, and could be checked with [IsValid](https://pkg.go.dev/net/netip#Addr.IsValid) to see if they are the zero value for a `netip.Addr`.

Looking at the function, it seems like it will only return, at most, one IPv4 address, and one IPv6 address. Having two maps with only one key set in each seems a bit overkill for what could be a single map:

```go
addrs := map[string]string{
    "ipv4": "10.10.10.10",
    "ipv6": "fd00::",
}
```

If these IP addresses are only ever going to be put into DNS records, would it make sense for `getMachinePublicIPs` to return a type (new or existing) that could be directly passed to `CreateUpdateDomainRecord`?

> **NOTE**: I know it's outside of the scope for this pull request, so maybe in another one, we could address that `CreateUpdateDomainRecord` takes 6 arguments (excluding the `context.Context`)! That might be better served by passing in a struct.

Reply:
I've simplified the code using a struct now. Still using a slice, but a slice of the struct, instead of a map